### PR TITLE
Songs collections

### PR DESCRIPTION
**Pull request by LatidoReMe**
JQuery and minor CSS for https://github.com/MagiCircles/SchoolIdolTomodachi/pull/35

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -17,8 +17,63 @@ function onVersionChange(form, animation) {
...
+        else{
+        form.find('#id_b_side_master').val('');
+        form.find('#id_b_side_start').val('');
+        form.find('#id_b_side_end').val('');
+        }
+}
+
+function loadSongs() {
+    let form = $('[data-form-name$="_song"]');
+    console.log(form);
```
Comment (db0company) [inline js/main.js:75]: Delete all `console.log` before merging.

File: js/main.js [hunk 2]
```diff
@@ -17,8 +17,63 @@ function onVersionChange(form, animation) {
...
+    hideExtraVersions(form);
+    onLocationChange(form);
+    form.find('#id_c_locations').change(function () { onLocationChange(form)});
```
Comment (db0company) [inline js/main.js:78]: Space + trailing column:
`onLocationChange(form)});` -> `onLocationChange(form); });`

File: less/style.less [hunk 3]
```diff
@@ -68,3 +68,8 @@ body {
         line-height: 100px;
     }
 }
+
+.songwriters-details {
+    display: inline-block;
+    text-align: left;
+}
```
Comment (db0company) [inline less/style.less:75]: If we go for the stars design I mentioned in the other PR, you'll need to add the CSS for it. Here's the current one on Sukutomo, which probably needs to be adapted:

```css
.song {
    .level-stars {
	.flaticon-star {
	    &:nth-child(-n+3) { color: #d8fd53; }
	    &:nth-child(n+4):nth-child(-n+6) { color: #f7fb34; }
	    &:nth-child(n+7):nth-child(-n+9) { color: #fbaac3; }
	    &:nth-child(n+10) { color: #f33256; }
	}
    }
}
```

File: js/main.js [hunk 4]
```diff
@@ -17,8 +17,63 @@ function onVersionChange(form, animation) {
...
+            field.hide('slow');
+            input.val('');
+    }
+    });
+}
+
+function onLocationChange(form){
+        let checkbox_h=form.find('#id_c_locations [value="hits"]');
+        let checkbox_d=form.find('#id_c_locations [value="daily"]');
+        let checkbox_b=form.find('#id_c_locations [value="bside"]');
```
Comment (db0company) [inline js/main.js:41]: Use explicit variable names: `checkbox_h` -> `checkbox_hits`.

Spaces around `=`:
```javascript
let checkbox_hits = form.find(...);
```

Comment (db0company) [reply js/main.js:41]: Maybe you can simplify the whole logic of this function by using a foreach that goes through the variables and their related fields:

```javascript
$.each({
    hits: ['unlock'],
    bside: ['b_side_master', 'b_side_start', 'b_side_end'],
    ...
}, function(checkbox_name, related_fields) {
    let checked = form.find('#id_c_locations [value="' + checkbox_name + '"]').prop('checked');
    $.each(related_fields, function(i, field_name) {
        let input = form.find('#id_' + field_name);
        let field = input.closest('form-group');
        if (checked) {
            field.show('fast');
        } else {
            input.val('');
            field.hide('fast');
        }
    });
});
```

I didn't actually check that the code runs, you'll need to check that 😅

Also you could make it easier to update in the future by providing the values in python:

```python
# models.py:

class Song(...):
    ...
    LOCATIONS = [
        ('hits', {
            'translation': _('Hits'),
            'related_fields': ['unlock'],
        }),
        ...
    ]
    LOCATIONS_CHOICES = [(_name, _info['translation']) for _name, _info in LOCATIONS.items()]
    LOCATIONS_RELATED_FIELDS = [(_name, _info['related_fields']) for _name, _info in LOCATIONS.items()]
    c_locations = models.TextField(_('Locations'), blank=True, null=True)

# magicollections.py:

    def _modification_extra_context(self, context):
        ...
        context['js_variables']['locations_related_fields'] = jsv(models.Song.LOCATIONS_RELATED_FIELDS)
```

Then you can just do:

```javascript
$.each(locations_related_fields, function(checkbox_name, related_fields) {
    ...
```

**Review by db0company — CHANGES_REQUESTED** [review]

File: js/main.js (hunk 1 above)
Comment (LatidoReMe) [reply js/main.js:75]: ✅

File: js/main.js (hunk 2 above)
Comment (LatidoReMe) [reply js/main.js:78]: ✅

File: less/style.less (hunk 3 above)
Comment (LatidoReMe) [reply less/style.less:75]: I have another PR currently which does something a bit different. I still can go for something like this if you prefer, but I think that it'll look best as just text in the end.

File: js/main.js (hunk 4 above)
Comment (LatidoReMe) [reply js/main.js:41]: The code needed to be adjusted, but I got a more loop-y version working! ✅

**Comment by LatidoReMe** [comment]
Closing due to #5

**Pull request closed without merging**